agent/agent.go

```diff
@@ -698,6 +710,109 @@ func (agent *Agent) handleGetMongoSummary() (interface{}, []error) {
 	return runRealCmd("pt-mongodb-summary")
 }
 
+func (agent *Agent) handleCollectInfo() (interface{}, []error) {
+	tmpDir, err := ioutil.TempDir("", "data_collection")
+	if err != nil {
+		return nil, []error{errors.Wrap(err, "cannot create temp dir for data collection")}
+	}
+
+	outFiles := []string{}
+	cmds := []string{"pt-summary", "pt-mysql-summary"}
+
+	for _, cmd := range cmds {
+		outFile := path.Join(tmpDir, cmd+".out")
+		outFiles = append(outFiles, outFile)
+		_, err := runRealCmd(cmd, "> "+outFile)
```

Comment:
Maybe I'm missing something, but how do you pass user/host/pass to tools?
https://github.com/percona/qan-agent/blob/master/query/plugin/mysql/summary/summary.go#L27-L45
https://github.com/percona/qan-agent/blob/master/query/plugin/mongo/summary/summary.go#L25-L41

Reply:
I am using the same methods used above. Is that incorrect?